include/onnxruntime/core/providers/cuda/cuda_provider_factory.h

```diff
@@ -10,7 +10,7 @@ extern "C" {
 /**
  * \param device_id cuda device id, starts from zero.
  */
-ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_CUDA, _In_ OrtSessionOptions* options, int device_id);
+ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_CUDA, _In_ OrtSessionOptions* options, int device_id, size_t cuda_mem_limit);
```

Comment:
This is an ABI breaking change. We shouldn't do such things. If you need an exception, Please send out an email to acknowledge all the people.

Reply:
Also mentioned [here](https://aiinfra.visualstudio.com/Lotus/_git/onnxruntime/pullrequest/5509?_a=files&path=%2Finclude%2Fonnxruntime%2Fcore%2Fproviders%2Fcuda%2Fcuda_provider_factory.h&discussionId=52724). There's a [task](https://msdata.visualstudio.com/Vienna/_boards/board/t/Training%20Runtime/Features/?fullScreen=true) to fix this. *We cannot break the ABI; no exceptions*.

Reply:
If that, I would suggest we change the "int device_id" to a list of "int device_id" at the same time.

Reply:
reverted per discussion with Sherlock